File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        counts = {}
        for i in nums:
            if i not in counts:
                counts[i] = 0
            counts[i] += 1
        sorted_items = sorted(counts.items(), key=lambda x: x[1], reverse=True)
        result = [num for num, freq in sorted_items[:k]]
        return result

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;    

        for (int n: nums) {
            freq[n]++;
        }

        vector<int> result;

        while(k--) {
            int bestKey = 0;
            int bestFreq = -1;


            for(auto &p: freq){
                if (p.second > bestFreq) {
                    bestFreq = p.second;
                    bestKey = p.first;
                }
            }
            result.push_back(bestKey);
            freq.erase(bestKey);
        }
        return result;
    }
};